pkg/profile/git_repo.go

```diff
@@ -1,25 +1,25 @@
 package profile
 
 import (
-	"strings"
+	"fmt"
 
 	sourcev1 "github.com/fluxcd/source-controller/api/v1beta1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func (p *Profile) makeGitRepository() *sourcev1.GitRepository {
+func (p *Profile) makeGitRepository(path string) *sourcev1.GitRepository {
 	ref := &sourcev1.GitRepositoryRef{
-		Branch: p.subscription.Spec.Branch,
+		Branch: fmt.Sprintf("%s:%s:%s", p.definition.Name, path, p.subscription.Spec.Branch),
 	}
 	if p.subscription.Spec.Tag != "" {
 		ref = &sourcev1.GitRepositoryRef{
-			Tag: p.subscription.Spec.Tag,
+			Tag: fmt.Sprintf("%s:%s:%s", p.definition.Name, path, p.subscription.Spec.Tag),
 		}
 	}
```

Comment:
slightly confused about this, are we just using this resource as a convenient way to pass information around?